### PR TITLE
Working with bash arrays

### LINKED ISSUE #7687: Working with bash arrays
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission do NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community members to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error-free). 

All revisions and edits should be *completed on your forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

**NOTE: Any topic suggestion copied/taken/modified from YouTube may Lead to a permanent ban from the community.**

### Topic Suggestion
Working with bash arrays

### Writing sample(s): 
https://www.section.io/engineering-education/concurrency-in-go/
https://www.section.io/engineering-education/integrating-graphql-api-in-a-django-application/

#### Proposed title of the article
Working with bash arrays

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Bash scripting, just like with other programming languages and scripting tools, allows you to create variables, run for loops, apply conditional logic, and more importantly, you can work with arrays, which are our focus in the tutorial. Arrays in Bash scripting are quite powerful for storing multiple data and fundamental data structures.

In this tutorial, we will walk you through what bash arrays are, the distinction between bash arrays and other concepts, and how to perform key operations on them with detailed examples.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
1. Good understanding of bash arrays.
2. How to perform key operations using bash arrays with detailed examples.
3. How to differentiate bash arrays from other similar bash concepts.
4. The different types of bash arrays with examples and code snippets.

#### Article quality
This article provides a detailed view of what bash arrays are and how to perform key operations with them using code snippets and examples. There are other concepts which may seem similar to bash array this article differentiates these concepts with examples. Lastly, shows how to add, remove and print elements in a bash array.

#### References
N/A.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquote notes as you fill in the form before you submit.
We look forward to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Anitaachu**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #7687`

### CODE REVIEW

**Comment by hectorkambow** [comment]
Our team has been working diligently to clear the backlog and we should be getting this pull request ready to be published next week 🚀

**Comment by WanjaMIKE** [comment]
Plagiarism check is done

**Comment by WanjaMIKE** [comment]
Sure, he can take it over @

**Comment by hectorkambow** [comment]
👍  Thank you @WanjaMIKE  
@Youngshark3  

Be sure to reach out if you have any questions along the way - be sure to use the comment feature via GitHub to leave notes.

**Comment by Youngshark3** [comment]
> 

okay thanks @hectorkambow on it

File: content/articles/working-with-bash-arrays/index.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+excerpt_separator: <!--more-->
+images: 
+
+  - url: /engineering-education/working-with-bash-arrays/hero.png
+    alt: Bash image
+---
+
+As a programmer, you have probably come across or used the command line for building an application, a tool, or scripting. Bash is likely one of the most common command-line interpreters used for writing scripts.
+
+Bash scripting, just like with other programming languages and scripting tools allows you to create variables, run for loops, apply conditional logic, and more importantly, you can work with arrays which are our focus in the tutorial. Arrays in bash scripting are quite powerful for storing multiple data and fundamental data structures.
```
Comment (Youngshark3) [inline content/articles/working-with-bash-arrays/index.md:20]: i find this sentence too long break it down to increase readability

File: content/articles/working-with-bash-arrays/index.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+# Declare the array
+declare -a new_array=("orange", "pear", "grape", "apple")
+
+# Initialize the array
+new_array=("Fruits")
+
+# Add a new variable
+new_array[4]=("Apple")
+```
+
```
Comment (Youngshark3) [inline content/articles/working-with-bash-arrays/index.md:116]: these codes are not inorder kindly edit them

**Review by Youngshark3 — CHANGES_REQUESTED** [review]

**Comment by Youngshark3** [comment]
@Anitaachu okay checking

**Comment by Youngshark3** [comment]
cc @WanjaMIKE @hectorkambow done its okay for the next step

**Comment by hectorkambow** [comment]
Did you complete the prelim review - or the tech review portion?? @Youngshark3

**Comment by hectorkambow** [comment]
If both are complete - we can add the "ready for final review" 👍

**Comment by Youngshark3** [comment]
@hectorkambow  yeah everything is fine. Done both prelim and tech review

**Comment by hectorkambow** [comment]
🚀  

Fantastic !!! We will be sure to give this a final review and assess the PR 👏  

Nicely done.

**Comment by hectorkambow** [comment]
👏  

Nicely done @Anitaachu  & @Youngshark3 

🚀  

Please be sure to go over any edits made to the PR to improve future blog posts and reviews. 


@Anitaachu 
You'll be able to see your live published article on the site (in about 30 mins or so).

To give your article a boost in page engagement - _**be sure to share it across your social site (LinkedIn/Twitter) and tag the EngEd program so we can also share it and give it some traction/exposure as well!**_

**If you have not done so already - consider adding a skill section to your author page. This will go a long way for potential future employers.**

>NOTE: Also, take into consideration that you can add any other past projects of yours: videos, talks, tutorials, articles (on ANY >other site) - you can add them to your EngEd profile.
>
>Think of the EngEd profile as your one-stop shop for everything you do in DEV - making it easier for you to share all your >accomplishments when you go out for an interview or employment opp.
>
>It can be from any blog site (or site in general).

Let me know if you have any questions moving forward.

**Comment by Youngshark3** [comment]
Thanks @hectorkambow and congratulations 🎉@Anitaachu

**Comment by hectorkambow** [comment]
@Anitaachu 
Hello hello! 

We wanted to announce that we're going to take a focus on content centered around Kubernetes (K8s) for the month of August.

We have enjoyed reading all amazing content that has been created the last few months - and with so much to explore with K8s we wanted to invite the community to approach it.

**Wit_h that said, the best five (5) K8s related articles that come in this month will be eligible to receive a complimentary virtual pass to [KubeCon USA](https://events.linuxfoundation.org/kubecon-cloudnativecon-north-america/) ($75.00 value) in Oct.
We're looking for content that enriches developers' understanding of K8s, best practices, and its many use cases.**_

Please be sure to go over the already [published content the EngEd community has created on Kubernetes](https://www.section.io/engineering-education/search/?q=kubernetes) to see where we can build upon or further explore.

While we're not restricting content solely to K8s-related articles over the next month, we will prioritize acceptance and review of K8s content over other topics.

Please be sure to complete the [topic suggestion form](https://github.com/section-engineering-education/engineering-education/issues/new?assignees=&labels=topic+suggestion&template=enged-content-idea-suggestion.md&title=) when you're ready to dive in.

Happy coding!

**Pull request merged**